### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by drbacke** [Akkudoktor-EOS/EOS#100] [General]
I think WE should start with a better documentation.
Any suggestions on how WE should do this?
Wiki, readme, API doc ...

**Comment by sandboxcode** [comment]
We document all internal repos and projects with a single [Nextra repo](https://nextra.site/) within our working groups Github organization (though it is private and I can't share it at this moment).

Great tool and if we want to use it, please open another repo and call it something like `EOS-docs`.
It makes it way easier to maintain.

**Comment by fortrieb** [comment]
Currently we can stick to README and if we have enough content to start a website then can starting with github pages.

**Comment by BerndCzech** [comment]
Is there a reason to keep the `README-DE.md`? Imo its duplication leading to confusion and increased effort with limited value.

I would offer to:
- [ ] check for inconsistencies
- [ ] merge all into `README.md`
- [ ] delete `README-DE.md`
ok?

**Reply by fortrieb** [reply]
I think translation of documentation make a lot of sense, to attract more users later. Currently project isn't at this state to be used by none developers. Should be ok to delete the german readme file.

**Comment by b0661** [comment]
What about [Sphinx](https://www.sphinx-doc.org/en/master/) and/or [Doxygen](https://www.doxygen.nl/) to extract documentation from source code and also add pure documentation files. The tools are used by many open source projects and can generate HTML and a lot of other formats. It´s also easy to upload the documentation to [Read the Docs](https://about.readthedocs.com/), which spares an own documentation server, provides good visibility and is free for open source projects.

Both tools ( [Sphinx](https://www.sphinx-doc.org/en/master/), [Doxygen](https://www.doxygen.nl/) ) support API documentation extraction from source code and enable to write documentation in [Markdown](https://www.sphinx-doc.org/en/master/usage/markdown.html) or [reStructuredText (reST)](https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html). Using such a simplified markup language for documentation enables to easily handle the documentation files in git with the same handling like source code files.

**Reply by fortrieb** [reply]
Extracting comments from source files isn't enough to get started with this project. Some tutorials would be helpful to get started by normal users if project reached a usable state.

**Reply by drbacke** [reply]
Yes If we have a First docker Release + swagger (usefull API documentation). 
This could be the start for the next step

**Comment by mattmeye** [comment]
Just another hint from a developer: Mkdocs Material, can also be easily integrated in github actions
https://squidfunk.github.io/mkdocs-material/publishing-your-site/?h=github#with-github-actions

**Comment by TheSmartGerman** [comment]
In my opinion documentation should be made easy accessible. 

Using/Enable the GitHub Wiki and make it editable for everyone. So everyone can contribute.

Try to start with a good structure.... 

e.g. I started this wiki (my part of contributing to this project) and it's growing since: https://github.com/open3e/open3e/wiki

**Reply by mattmeye** [reply]
I completely agree with you. The focus should be on documentation (preferably in Markdown) and you shouldn't get bogged down in visualization/publication.

As I said, I often use mkdocs material both privately and professionally with large development teams. You define where the documents are located, where the generated documentation should be located, and that's it. In my opinion, the biggest feature is the search across all documents.

**Reply by mattmeye** [reply]
I played a little bit https://mattmeye.github.io/EOS/welcome/ and just found out that it exists also a sphinx variant https://jbms.github.io/sphinx-immaterial/index.html
